### ISSUE TITLE
Non-atomic transaction when yielding native Promise

### ISSUE THREAD

**Issue by ThomasGreiner** [dfahlander/Dexie.js#444]
I've encountered an issue where the order of operations is unreliable and distilled it into this example:

```html
<script src="https://unpkg.com/dexie@latest/dist/dexie.js"></script>
<script>
"use strict";

let db = new Dexie("db");
db.version(1).stores({items: "&key,value"});

function set(key, value) {
  return db.transaction("rw", db.items, function*() {
    yield Promise.resolve();
    
    let item = yield db.items.where("key").equals(key).first();
    if (!item) {
      yield db.items.add({key, value});
    } else {
      yield db.items.update(key, {value});
    }
  });
}

set("foo", "bar");
set("foo", "baz");
</script>
```

The resulting database entry is `{key: "foo", value: "bar"}` and the following error is thrown:
```
Unhandled rejection: ConstraintError: Key already exists in the object store.
```

The expected result `{key: "foo", value: "baz"}` can, however, be achieved by removing the line `yield Promise.resolve();`.

**Comment by dfahlander** [comment]
As pointed out on various places in the documentation pages, only Dexie.Promise can be used within transactions in Dexie 1.x.

If you use Dexie 2.0.0 (still in beta), you don't need to think about this anymore. Try upgrade dexie to dexie@next and see if it solves the issue.

**Comment by ThomasGreiner** [comment]
Thanks, I'll try it out whether I can use `Dexie.Promise` as a wrapper. I checked out various places in the documentation such as the README and the "Simplify with yield" page and the "Dexie.transaction()" page and didn't find this mentioned anywhere, nor did I find examples which yielded Promises not coming from database operations. But reading more into the documentation of "Dexie.transaction()" I noticed now that it was mentioned in there so sorry that I missed that.

I can confirm that using `dexie@next` works so thanks for the advice.
